tests/integration/test_end_to_end.py

```diff
@@ -99,6 +107,15 @@ def test_end_to_end(capsys, fake_dns, fake_get_config, openstack_session):
         instance_name[-2:]
     )
 
+    # check ebs created
+    time.sleep(10)
+    volume = openstack_session.compute_session.list_volumes()[0]
```

Comment:
this formulation is slightly unusual. if something is wrong, the assignment will raise `IndexError`, whereas your assert below is unlikely to ever fail, unless `list_volumes` returns e.g. `[False]` or `[None]`.

more common might be
`assert s.list_volumes()  # makes sure this isn't empty`

or even 
```
volumes = s.list_volumes())
assert len(volumes) == 1
```

and then just
`volume = volumes()[0]`